src/test/functions/projectConfig.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { expect } from "chai";
+
+import * as projectConfig from "../../functions/projectConfig";
+import { FirebaseError } from "../../error";
+
+const TEST_CONFIG = { source: "foo" };
+
+describe("projectConfig", () => {
+  describe("normalize", () => {
+    it("normalizes singleton config", () => {
+      expect(projectConfig.normalize(TEST_CONFIG)).to.deep.equal([TEST_CONFIG]);
+    });
+
+    it("normalizes array config", () => {
+      expect(projectConfig.normalize([TEST_CONFIG])).to.deep.equal([TEST_CONFIG]);
+    });
+
+    it("throws error if given empty config", () => {
+      expect(() => projectConfig.normalize([])).to.throw(FirebaseError);
+    });
+  });
+
+  describe("validate", () => {
+    it("passes validation for simple config", () => {
+      expect(projectConfig.validate([TEST_CONFIG])).to.deep.equal([TEST_CONFIG]);
+    });
+
+    it("fails validation given more than one config", () => {
+      expect(() =>
+        projectConfig.validate([TEST_CONFIG, { ...TEST_CONFIG, source: "bar" }])
```

Comment:
why don't you just write `projectConfig.validate([TEST_CONFIG, { source: "bar" }])`?